### PR TITLE
Canaan and Sara

### PR DESCRIPTION
**Pull request by canaanwest**
# Scrabble
Congratulations! You're submitting your assignment.

## Comprehension Questions

|  Feature 	|   Feedback	|
|---	|---	|
|   Why do you think we chose to make the `score` method in the `Scoring` class a **class method** instead of an **instance method**? |  Because we wanted to be able to access the scoring method without having to make a new instance of score each time!  |
|   Describe an example of a test you wrote for a _nominal case_. | "It Can recognize a letter's value and add it to the score" -- this accepts and checks valid input to make sure the function can properly score valid letters.  |
|   Describe an example of a test you wrote for an _edge case_. |  For the same scoring method, the edge case we used was when we tried to score a word that was too long or a word that contained numbers. 	|
|   What was your final test coverage percentage?	|   98.73%	|
|   Is there a specific piece of code you'd like feedback on?	|   We ended up passing tilebag as a parameter for Player. We were very sad when this made our code incompatible with the game program provided.  We're still kind of confused about how else we could have checked the tilebag without needing a placeholder for it within player (without involving the Game file provided).	|

### CODE REVIEW

File: lib/player.rb [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    def initialize(name, tile_bag)
+      @name = name
+      @tile_bag = tile_bag
+      @word_scores = []
+      @plays = []
+      @hand = []
+    end
+
+    def play(word)
+      if won? == true
```
Comment (CheezItMan) [inline lib/player.rb:16]: You can instead do:

`if won?`

**Comment by CheezItMan** [comment]
# Scrabble
## What We're Looking For

|  Feature  |   Feedback |
|---	|---	|
|  **General**	|   	|
|   Answered comprehension questions  |  Check, with `draw_tiles`, you should pass the `Tilebag` object to the method as a parameter rather than have each `Player` have it's own `Tilebag`.  It doesn't make sense for each player to have their own `Tilebag`.  |
|   Both Teammates contributed to the codebase	| Good number of commits, both team members contributing.   |
|   Regular Commits with meaningful commit messages  |  Excellent  |
|  **Scoring class**	|    |
|  `score` method	|  Good use of a Regex  |
|   Uses appropriate data structure to store the letter score	| Rather than use a case statement I would use a hash where the keys are the letters and the values are the score to calculate a word score.    |
|   Appropriately handles edge cases	|  Handle invalid input and 7-letter-words.  |
|   Tests `score` edge cases (one-letter word, 7 letter word)	|  Check  |
|  `highest_scoring_word` method	|  Check, I would not return "None" if the array is empty.  Returning `nil` makes more sense.    |
|   Appropriately handles edge cases	|  Well done, good use of enumerables.    |
|   Tests `highest_scoring_word` edge cases (ties, no words, no array)	|   I would also test the arrays by changing the word order and verifying that ties, which match on length returns the first word.   |
|  **Player class**	|   	|
|  Uses an array to keep track of words played	|  Check  |
|  Uses existing code to create `highest_scoring_word` and `highest_word_score`	| Check   |
|  Returns `true` or `false` from `won?` method | check  |
|  Tests edge cases on the `play(word)` method for words that cannot be scored |   Yes through `score`.  |
|  **TileBag class**	|   	|
|  Uses a data structure to keep the set of default tiles in the bag	|  The hash works, but you have to reconstruct an array every time someone draws tiles.    |
|  `draw_tiles` method uses the parameter to draw a set of random tiles |  **No** instead it uses a `Tilebag` instance variables.    |
| `tiles_remaining` method returns a count	|  Check  |
|  Tests `draw_tiles` edge cases for different parameter inputs	|  There's no test to see what happens when you draw all the tiles in the Tilebag  |
|  Summary | Pretty well done, you needed to look at the instructions for `draw_tiles`  as it was supposed to take a parameter.  Nice work overall!  You hit all the requirements.  |

**Review by CheezItMan — COMMENTED** [review]
Nice work
